component/src/main/java/org/ballerinalang/data/cassandra/CassandraDataSource.java

```diff
@@ -45,32 +65,459 @@ public Session getSession() {
+        if (protocolOptionsConfig != null) {
+            populateProtocolOptions(builder, protocolOptionsConfig);
+        }
+        String clusterName = options.getStringField(ConnectionParam.CLUSTER_NAME.getIndex());
+        if (!clusterName.isEmpty()) {
+            builder.withClusterName(clusterName);
+        }
+        boolean jmxReportingDisabled = options.getBooleanField(ConnectionParam.WITHOUT_JMX_REPORTING.getIndex()) != 0;
+        if (jmxReportingDisabled) {
+            builder.withoutJMXReporting();
+        }
+        boolean metricsDisabled = options.getBooleanField(ConnectionParam.WITHOUT_METRICS.getIndex()) != 0;
+        if (metricsDisabled) {
+            builder.withoutMetrics();
+        }
+        populateLoadBalancingPolicy(builder, options);
+        populateReconnectionPolicy(builder, options);
+        populateRetryPolicy(builder, options);
+
+        return builder;
+    }
+
+    /**
+     * Populates the Cassnadra RetryPolicy options in the Cluster Builder.
+     *
+     * @param builder Cluster Builder
+     * @param options BStruct containing available options for cluster connection initialization
+     */
+    private void populateRetryPolicy(Cluster.Builder builder, BStruct options) {
+        String retryPolicyString = options.getStringField(ConnectionParam.RETRY_POLICY.getIndex());
+        if (!retryPolicyString.isEmpty()) {
+            RetryPolicy retryPolicy = retrieveRetryPolicy(retryPolicyString);
+            switch (retryPolicy) {
+            case DEFAULT_RETRY_POLICY:
+                builder.withRetryPolicy(DefaultRetryPolicy.INSTANCE);
+                break;
+            case FALLTHROUGH_RETRY_POLICY:
+                builder.withRetryPolicy(FallthroughRetryPolicy.INSTANCE);
+                break;
+            case DOWNGRADING_CONSISTENCY_RETRY_POLICY:
+                builder.withRetryPolicy(DowngradingConsistencyRetryPolicy.INSTANCE);
+                break;
+            case LOGGING_DEFAULT_RETRY_POLICY:
+                builder.withRetryPolicy(new LoggingRetryPolicy(DefaultRetryPolicy.INSTANCE));
+                break;
+            case LOGGING_FALLTHROUGH_RETRY_POLICY:
+                builder.withRetryPolicy(new LoggingRetryPolicy(FallthroughRetryPolicy.INSTANCE));
+                break;
+            case LOGGING_DOWNGRADING_CONSISTENCY_RETRY_POLICY:
+                builder.withRetryPolicy(new LoggingRetryPolicy(DowngradingConsistencyRetryPolicy.INSTANCE));
+                break;
+            default:
+                throw new UnsupportedOperationException(
+                        "Support for the retry policy \"" + retryPolicy + "\" is not implemented yet");
+            }
+        }
+    }
+
+    private RetryPolicy retrieveRetryPolicy(String retryPolicy) {
+        try {
+            return RetryPolicy.fromPolicyName(retryPolicy);
+        } catch (IllegalArgumentException e) {
+            throw new BallerinaException("\"" + retryPolicy + "\"" + " is not a valid retry Policy");
```

Comment:
Shall we stick to one of "policy" or "Policy" in all the places?

Reply:
+1